plugin-niassa+pinery/src/main/java/ca/on/oicr/gsi/shesmu/niassa/WorkflowAction.java

```diff
@@ -536,6 +536,14 @@ public synchronized boolean performCommand(String commandName) {
           attribute.setTag("skip");
           attribute.setValue("shesmu-ui");
           server.get().metadata().annotateWorkflowRun(runAccession, attribute, null);
+          final WorkflowRun run = server.get().metadata().getWorkflowRun(runAccession);
+          switch (run.getStatus()) {
+            case running:
+            case pending:
+            case submitted:
+              run.setStatus(WorkflowRunStatus.cancelled);
```

Comment:
This probably should be submit_cancel to let the propagate statuses job kill the workflow run.

Reply:
Fixed. I really don't understand how these statuses work.